library/src/main/java/com/datatorrent/lib/fileaccess/FileAccess.java

```diff
@@ -151,7 +151,7 @@
      * @param value
      * @throws IOException
      */
-    void append(byte[] key, byte[] value) throws IOException;
+    void append(Slice key, Slice value) throws IOException;
```

Comment:
We will modify existing implementations/utilities in Malhar to use the new ones anyways, but we don't break anything outside of Apex by deprecating old ones.

My point is that this is an old interface, and deprecating just one method and adding a new one, doesn't seem to cause much disruption. If HDHT moves to Malhar some day, having the old deprecated method may make the move easy. I don't have any strong opinion on this. However, it will be helpful to seek guidance from @tweise on this.